deepchem/molnet/load_function/zinc15_datasets.py

```diff
@@ -2,59 +2,68 @@
-
-# Names of supported featurizers
-zinc15_featurizers = [
-    'SmilesToImage', 'OneHotFeaturizer', 'SmilesToSeq', 'RDKitDescriptors',
-    'ConvMolFeaturizer', 'WeaveFeaturizer', 'CircularFingerprint',
-    'Mol2VecFingerprint'
-]
-DEFAULT_FEATURIZERS = {k: DEFAULT_FEATURIZERS[k] for k in zinc15_featurizers}
-
-# dict of accepted transformers
-DEFAULT_TRANSFORMERS = get_defaults("trans")
-
-# dict of accepted splitters
-DEFAULT_SPLITTERS = get_defaults("splits")
-
-# names of supported splitters
-zinc15_splitters = ['RandomSplitter', 'RandomStratifiedSplitter']
-DEFAULT_SPLITTERS = {k: DEFAULT_SPLITTERS[k] for k in zinc15_splitters}
+import deepchem as dc
+from deepchem.molnet.load_function.molnet_loader import TransformerGenerator, _MolnetLoader
+from deepchem.data import Dataset
+from typing import List, Optional, Tuple, Union
+
+ZINC15_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/"
+ZINC15_TASKS = ['mwt', 'logp', 'reactive']
+
+
+class _Zinc15Loader(_MolnetLoader):
+
+  def __init__(self, *args, dataset_size: str, dataset_dimension: str,
+               **kwargs):
+    super(_Zinc15Loader, self).__init__(*args, **kwargs)
+    self.dataset_size = dataset_size
+    self.dataset_dimension = dataset_dimension
+    self.name = 'zinc15_' + dataset_size + '_' + dataset_dimension
+
+  def create_dataset(self) -> Dataset:
+    if self.dataset_size not in ['250K', '1M', '10M', '270M']:
+      raise ValueError(
+          "Only '250K', '1M', '10M', and '270M' are supported for dataset_size."
+      )
+    if self.dataset_dimension != '2D':
+      raise ValueError(
+          "Currently, only '2D' is supported for dataset_dimension.")
+    if self.dataset_size == '270M':
+      answer = ''
+      while answer not in ['y', 'n']:
+        answer = input("""You're about to download 270M SMILES strings.
+          This dataset is 23GB. Are you sure you want to continue? (Y/N)"""
+                      ).lower()
+      if answer == 'n':
+        raise ValueError('Choose a smaller dataset_size.')
+    filename = self.name + '.csv'
+    dataset_file = os.path.join(self.data_dir, filename)
+    if not os.path.exists(dataset_file):
+      compressed_file = self.name + '.tar.gz'
+      if not os.path.exists(compressed_file):
+        dc.utils.download_url(
+            url=ZINC15_URL + compressed_file, dest_dir=self.data_dir)
+      dc.utils.untargz_file(
+          os.path.join(self.data_dir, compressed_file), self.data_dir)
+    loader = dc.data.CSVLoader(
+        tasks=self.tasks,
+        feature_field="smiles",
+        id_field="zinc_id",
+        featurizer=self.featurizer)
+    return loader.create_dataset(dataset_file, shard_size=8192)
 
 
 def load_zinc15(
-    featurizer=DEFAULT_FEATURIZERS['OneHotFeaturizer'],
-    transformers: List = [DEFAULT_TRANSFORMERS['NormalizationTransformer']],
-    splitter=DEFAULT_SPLITTERS['RandomSplitter'],
+    featurizer: Union[dc.feat.Featurizer, str] = dc.feat.OneHotFeaturizer(),
```

Comment:
This line also requires RDKit as a hard requirement and leads to the failure of importing deepchem.